lib/composition.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'extensions/vector2.dart';
+
+class Composition {
+  /// The images that are going to be composited together.
+  final List<Image> _images = [];
+
+  /// The positions where each [_images] will be added.
+  final List<Vector2> _positions = [];
+
+  /// The sources that should be used on each [_images].
+  final List<Rect> _sources = [];
+
+  /// The [BlendMode] to be used for each [_images].
+  final List<BlendMode> _blendModes = [];
```

Comment:
would it be clearer to have a single list of a data class (Fragment or something) that has all these fields?

Reply:
I didn't see a need to introduce another class just for that. But atm you also can't clear the composition, so if we introduce that then it would be smarter to have a single list.

Reply:
I added a `clear` method and refactored it to use a single list.